auto_submit/lib/service/validation_service.dart

```diff
@@ -197,8 +196,15 @@ class ValidationService {
     final GithubService githubService = await config.createGithubService(slug);
 
     if (revertValidationResult.result) {
-      ApproverService approverService = ApproverService(config);
-      await approverService.approve(messagePullRequest);
+      // Approve the pull request automatically as it has been validated.
```

Comment:
Maybe we can reuse the code on the approver service. Something like:

making `Future<void> approve(PullRequest pullRequest) async {` private removing the logic to check the rollers accounts, and then creating two wrapper methods: autoApproval that validates the autoroller accounts validation and then call the private approve and then revertApproval with the check for revert and then delegating to approve.

Reply:
Okay that sounds good. I will look into that.